processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -254,6 +254,38 @@ public File merge(
     return merge(indexes, metricAggs, outDir, segmentMetadata, indexSpec, new BaseProgressIndicator());
   }
 
+  private List<String> getLexicographicMergedDimensions(List<IndexableAdapter> indexes)
+  {
+    return mergeIndexed(
+        Lists.transform(
+            indexes,
+            new Function<IndexableAdapter, Iterable<String>>()
+            {
+              @Override
+              public Iterable<String> apply(@Nullable IndexableAdapter input)
+              {
+                return input.getDimensionNames();
+              }
+            }
+        )
+    );
+  }
+
+  private List<String> getMergedDimensions(List<IndexableAdapter> indexes)
+  {
+    if (indexes.size() == 0) {
+      return ImmutableList.of();
+    }
+    List<String> dimOrder = ImmutableList.copyOf(indexes.get(0).getDimensionNames());
+    for (IndexableAdapter index : indexes) {
+      List<String> dimOrder2 = ImmutableList.copyOf(index.getDimensionNames());
+      if (!dimOrder.equals(dimOrder2)) {
+        return getLexicographicMergedDimensions(indexes);
+      }
+    }
```

Comment:
is copying the dimensionNames list necessary here ? I guess, it only needs to be done for the return statement.